### ISSUE TITLE
[consistent-type-definitions] Add a way to use interfaces in module declarations when rule is enforcing type aliases.

### ISSUE THREAD

**Issue by RebeccaStevens** [typescript-eslint/typescript-eslint#3792]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/consistent-type-definitions": ["error", "type"]
  }
}
```

```TS
declare module "fp-ts/HKT" {
  interface URItoKind2<E, A> {
    readonly ["SomeHKT"]: SomeType<E, A>;
  }
}
```

**Expected Result**

For this particular type definition, an type alias cannot be used. There should be a way to tell the rule to ignore interfaces inside module declarations.

**Actual Result**

There is not way to exclude this interface (other than eslint-disable).

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.29.2` |
| `@typescript-eslint/parser`        | `4.29.2` |
| `TypeScript`                       | `4.4.1-rc` |
| `ESLint`                           | `7.32.0` |
| `node`                             | `16.4.0` |
